File: backend/App/Domain/ValueObjects/ResetCode.php
<?php

declare(strict_types=1);

namespace App\Domain\ValueObjects;

use App\Domain\ValueObjects\Base\IntegerField;

final readonly class ResetCode extends IntegerField
{
    private const int FIXED_CODE_DIGITS = 6;
    protected const int MIN_VALUE = 10 ** (self::FIXED_CODE_DIGITS - 1);
    protected const int MAX_VALUE = 10 ** self::FIXED_CODE_DIGITS - 1;

    public static function getNewCode(): int
    {
        return mt_rand(self::MIN_VALUE, self::MAX_VALUE);
    }
}

File: tests/domain/value_objects/reset_code.py
from dataclasses import dataclass

from domain.value_objects.base import IntegerField

_FIXED_CODE_DIGITS = 6


@dataclass(slots=True, frozen=True)
class ResetCode(IntegerField):
    _min_value = 10 ** (_FIXED_CODE_DIGITS - 1)
    _max_value = 10**_FIXED_CODE_DIGITS - 1
